addons/pos_sms/static/src/js/receipt_screen.js

```diff
@@ -0,0 +1,74 @@
+/** @odoo-module */
+
+import { ReceiptScreen } from "@point_of_sale/app/screens/receipt_screen/receipt_screen";
+import { useTrackedAsync } from "@point_of_sale/app/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
+import { patch } from "@web/core/utils/patch";
+
+patch(ReceiptScreen.prototype, {
+    setup() {
+        super.setup(...arguments);
+        const partner = this.currentOrder.get_partner();
+        this.state.inputValue =
+            partner && partner.email
+                ? partner.email
+                : partner && partner.mobile
+                ? partner.mobile
+                : partner && partner.phone
+                ? partner.phone
+                : "";
+        this.doSendSms = useTrackedAsync(() => this._sendSmsReceiptToCustomer());
+    },
+
+    get is_valid_mobile() {
+        const value = this.state.inputValue;
+        if (value) {
+            const isValidFormat = /^[\d()+-\s]*$/.test(value);
+            const validMobile = value.replace(/[^\d]/g, "");
+            return isValidFormat && validMobile.length > 8 && validMobile.length < 15;
+        }
+        return false;
+    },
+
+    get smsNotice() {
+        switch (this.doSendSms.status) {
+            case "loading":
+                return { class: "text-info", message: _t("Sending in progress.") };
+            case "success":
+                return { class: "successful text-success", message: _t("SMS sent.") };
+            case "error":
+                return {
+                    class: "failed text-danger",
+                    message: _t("Sending SMS failed. Please try again."),
+                };
+            default:
+                throw new Error("Shouldn't be reached.");
+        }
+    },
+
+    _fetchMessageStatus() {
+        if (this.state.messageMode === "sms") {
+            return {
+                class: this.smsNotice.class,
+                message: this.smsNotice.message,
+                status: this.doSendSms.status,
+            };
+        }
+        return super._fetchMessageStatus();
+    },
+
+    onInputChange(ev) {
+        super.onInputChange(...arguments);
+        this.state.inputValue = ev.target.value;
+    },
```

Comment:
not needed by using t-model

Reply:
@vlst-odoo 
All changes are done, Can you please check again?
Thanks!